operator/roles/kiali-deploy/templates/kubernetes/service.yaml

```diff
@@ -9,7 +9,7 @@ metadata:
 spec:
   type: {{ kiali_vars.deployment.service_type }}
   ports:
-  - name: tcp
+  - name: {{ 'http' if kiali_vars.identity.cert_file == "" else 'tcp' }}
     protocol: TCP
```

Comment:
Does the Protocol needs to change to "HTTP" too?

Reply:
No, that's the layer 4 protocol

From https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.15/#serviceport-v1-core this can only be TCP, UDP, or SCTP